File: benchmarks/fixtures/cmdinj_fixture.py
# OS command injection (CWE-78) fixture (Python), modeled on ansys CVE-2024-29189.
import subprocess


def vulnerable(args):
    # spawned via a shell -> metacharacters in args execute
    return subprocess.Popen(args, shell=True)


def safe(args):
    # no shell; argv list passed directly
    return subprocess.Popen(args)

File: benchmarks/fixtures/cmdinj_fixture.js
// OS command injection (CWE-78) fixture: vulnerable + safe, one file. Detector only parses.
const { execSync, execFileSync } = require('child_process');

function vulnerable(userArg) {
  // dynamic input string-built into a shell command
  return execSync('npm ' + userArg);
}

function safe(userArg) {
  // argv array, no shell — the value is never re-parsed by a shell
  return execFileSync('npm', [userArg]);
}
